### PR TITLE
Separate out the actual work for FastqToSam so an outside caller can dir...

### PR DESCRIPTION
**Pull request by geoffjentry**
...ectly pass in FastqReaders

### CODE REVIEW

File: src/java/picard/sam/FastqToSam.java [hunk 1]
```diff
@@ -135,41 +165,62 @@ public static void main(final String[] argv) {
...
+
+        QUALITY_FORMAT = FastqToSam.determineQualityFormat(reader, reader2, QUALITY_FORMAT);
+
+        final SAMFileHeader header = createSamFileHeader();
+        final SAMFileWriter writer = new SAMFileWriterFactory().makeSAMOrBAMWriter(header, false, OUTPUT);
+
+        reader = fileToFastqReader(FASTQ);
+        if (FASTQ2 != null) {
+            reader2 = fileToFastqReader(FASTQ2);
+        }
```
Comment (jacarey) [inline src/java/picard/sam/FastqToSam.java]: Its kind of annoying that we have to open the readers to determine quality then close them.. them reopen new ones to start reading.

Comment (geoffjentry) [inline src/java/picard/sam/FastqToSam.java]: I agree, which was part of my problem in the first place as I was sucking the data through on a single named pipe. You could potentially tee the input or something like that but admittedly I was trying to do the minimum amount of change to enable what I needed

**Comment by geoffjentry** [comment]
Is that a :+1: @jacarey ?

**Comment by jacarey** [comment]
Yes sorry. (thumbsup)

**Pull request merged**